### PR TITLE
(DPS-1419) Copy to clipboard

### PR DESCRIPTION
**Pull request by ale-kacperczyk**
## 🚀 Links

- [Feature branch](https://developers.labs.livechat.com/docs/feature/dps-1419)
- [Jira](https://livechatinc.atlassian.net/browse/DPS-1419)

## 📓 Description
Added support for copy to clipboard in code snippets (CodeResponse and CodeSample, but only the ones that have a topbar).

image

image

## ✅ Checklist (for API docs)

- Changelog
- API versions
- Web & RTM
- **Table:** Available methods/webhooks/pushes
- **Table:** Specifics

For more guidelines, see [Updating API docs](https://livechatinc.atlassian.net/wiki/spaces/PAT/pages/761529400/Updating+API+docs).

### Features (for the website)

- New feature

### CODE REVIEW

File: content/messaging/agent-chat-api/v3.2/index.mdx [hunk 1]
```diff
@@ -89,6 +89,11 @@ The table shows scopes dependency of accessing chat parts:
 
 ### Agents scopes
 
+<Scopes
```
Comment (jakubsikora) [inline content/messaging/agent-chat-api/v3.2/index.mdx:92]: I dont think thats supposed to be here ?

File: content/messaging/agent-chat-api/v3.2/index.mdx [hunk 2]
```diff
@@ -89,6 +89,11 @@ The table shows scopes dependency of accessing chat parts:
```
Comment (jakubsikora) [inline content/messaging/agent-chat-api/v3.2/index.mdx:89]: Do you need these changes in this PR ?

File: src/components/extensions/CodeBlocks.js [hunk 3]
```diff
@@ -153,15 +170,18 @@ export const CodeSample = ({ path, children }) => {
...
-          {count > 1 && (
-            <SelectLanguage onChange={e => setSample(e.target.value)}>
-              {childrenArray.map(children => (
-                <option key={children.props.label}>
-                  {children.props.label}
-                </option>
-              ))}
-            </SelectLanguage>
-          )}
+          <CopyAndLanguageWrapper>
```
Comment (jakubsikora) [inline src/components/extensions/CodeBlocks.js:173]: I'm not sure if you need that wrapper here ?

File: src/components/extensions/CodeBlocks.js [hunk 4]
```diff
@@ -172,14 +192,19 @@ export const CodeSample = ({ path, children }) => {
...
+      )}
       <Body>{children}</Body>
     </CodeResponseWrapper>
   );
 };
 
 export const Code = ({ children }) => (
   <CodeWrapper>
-    <StickyWrapper>{children}</StickyWrapper>
+    <StickyWrapper>{children} </StickyWrapper>
```
Comment (jakubsikora) [inline src/components/extensions/CodeBlocks.js:208]: Why this space here ?

File: content/messaging/agent-chat-api/v3.2/index.mdx [hunk 5]
```diff
@@ -351,19 +351,19 @@ The **Message** event contains text message to other chat users.
```
Comment (jakubsikora) [inline content/messaging/agent-chat-api/v3.2/index.mdx:351]: Do you need this file in this PR ?

File: src/components/extensions/CopyToClipboardIcon.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+import styled from "@emotion/styled";
+import { CopyToClipboard } from "react-copy-to-clipboard";
+import { CopyIcon } from "../core/icons";
+import { logAmplitudeEvent } from "../../utils/index";
+
+const CopyToClipboardWrapper = styled.div`
+  position: relative;
+  flex-grow: 1;
+  min-width: 47px;
+  max-width: 50px;
```
Comment (jakubsikora) [inline src/components/extensions/CopyToClipboardIcon.js:11]:
```suggestion
  width: 50px;
```

that should do it

File: src/components/extensions/CopyToClipboardIcon.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+const CopyToClipboardAction = styled.div`
+  position: relative;
+  min-height: 18px;
+`;
+
+const CopyIconWrapper = styled.div`
+  position: absolute;
+  top: 0;
+  left: 20;
```
Comment (jakubsikora) [inline src/components/extensions/CopyToClipboardIcon.js:22]:
```suggestion
```

this is not needed

File: src/components/extensions/CopyToClipboardIcon.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    cursor: ${({ isCopied }) => (isCopied ? "initial" : "pointer")};
+  }
+`;
+
+const TextWrapper = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  text-align: right;
```
Comment (ale-kacperczyk) [inline src/components/extensions/CopyToClipboardIcon.js:38]:
```suggestion
  text-align: right;
  text-transform: capitalize;
```

**Pull request merged**